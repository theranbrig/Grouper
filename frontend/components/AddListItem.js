import gql from 'graphql-tag';
import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import { Button, Form, Message } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { INDIVIDUAL_LIST_QUERY } from './List';
import AddItemStyles from './styles/AddItemStyles';

const ADD_ITEM_MUTATION = gql`
  mutation ADD_ITEM_MUTATION($name: String!, $price: Int, $list: String!) {
    addItem(name: $name, price: $price, list: $list) {
      id
      name
      price
    }
  }
`;

class AddListItem extends Component {
  state = {
    name: '',
    price: 0,
    completed: false,
  };

  saveToState = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleDismiss = () => {
    this.setState({ completed: false });
  };

  render() {
    return (
      <Mutation
        mutation={ADD_ITEM_MUTATION}
        variables={{ list: this.props.id, name: this.state.name, price: this.state.price }}
        refetchQueries={[{ query: INDIVIDUAL_LIST_QUERY, variables: { id: this.props.id } }]}
      >
        {(addItem, { loading, error }) => (
          <AddItemStyles>
            <Form
              error={error}
              success={this.state.completed}
              loading={loading}
              inverted
              method="post"
              onSubmit={async e => {
                await addItem();
                this.setState({ name: '', price: 0, completed: true });
              }}
            >
              <Message onDismiss={this.handleDismiss} success header="Success!" content="Item Added to List" />
              <Message error header="Oops!" content={error && error.message.replace('GraphQL error: ', '')} />
              <Form.Group>
                <Form.Input
                  required
                  width={12}
                  type="text"
                  name="name"
                  id="name"
                  label="Item"
                  value={this.state.name}
                  minLength={5}
                  maxLength={50}
                  onChange={this.saveToState}
                  placeholder="Enter Item Name - 5 Char Min / 50 Char Max"
                />
                <Form.Input
                  className="number-input"
                  width={4}
                  id="price"
                  type="number"
                  name="price"
                  label="Max Price: 0 = No Max."
                  value={this.state.price}
                  onChange={this.saveToState}
                  placeholder="Optional - Enter Whole Dollar Amounts"
                  icon="dollar"
                  iconPosition="left"
                />
                <Button inverted icon="add" className="add-item-button" />
              </Form.Group>
            </Form>
          </AddItemStyles>
        )}
      </Mutation>
    );
  }
}

AddListItem.propTypes = {
  id: PropTypes.string.isRequired,
};

export default AddListItem;
