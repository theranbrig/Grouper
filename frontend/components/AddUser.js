import React, { Component } from 'react';
import { Button, Form } from 'semantic-ui-react';
import FormStyles from './styles/FormStyles';
import gql from 'graphql-tag';
import { Mutation, Query } from 'react-apollo';
import { INDIVIDUAL_LIST_QUERY } from './List';

const ADD_USER_MUTATION = gql`
	mutation ADD_USER_MUTATION($email: String!, $id: ID!) {
		addUser(email: $email, id: $id) {
			id
		}
	}
`;

class AddUser extends Component {
	state = {
		email: ''
	};

	saveToState = e => {
		this.setState({ [e.target.name]: e.target.value });
	};

	render() {
		return (
			<Mutation
				mutation={ADD_USER_MUTATION}
				variables={{ id: this.props.id, email: this.state.email }}
				refetchQueries={[{ query: INDIVIDUAL_LIST_QUERY, variables: { id: this.props.id } }]}>
				{(addUser, { loading, error }) => {
					if (error) <p>Error...</p>;
					return (
						<FormStyles>
							<Form
								inverted
								method="post"
								onSubmit={async e => {
									e.preventDefault();
									await addUser();
									this.setState({ email: '' });
								}}>
								<Form.Group>
									<Form.Input
										width={16}
										type="text"
										name="email"
										label="User Email"
										value={this.state.email}
										onChange={this.saveToState}
										placeholder="Enter New User Email"
									/>
								</Form.Group>
								<Button inverted type="submit">
									Add{loading ? 'ing' : ''} User
								</Button>
							</Form>
						</FormStyles>
					);
				}}
			</Mutation>
		);
	}
}

export default AddUser;
