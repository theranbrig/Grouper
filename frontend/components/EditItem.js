import gql from 'graphql-tag';
import Link from 'next/link';
import React, { Component } from 'react';
import { Mutation, Query } from 'react-apollo';
import { Button, Form, Grid, Icon, Message } from 'semantic-ui-react';
import AddItemStyles from './styles/AddItemStyles';
import Head from 'next/head';

const ITEM_QUERY = gql`
	query ITEM_QUERY($id: ID!) {
		listItem(where: { id: $id }) {
			id
			name
			price
			list
		}
	}
`;

const EDIT_ITEM_MUTATION = gql`
	mutation EDIT_ITEM_MUTATION($id: ID!, $name: String, $price: Int) {
		editListItem(id: $id, name: $name, price: $price) {
			id
			name
			price
			list
		}
	}
`;

class EditItem extends Component {
	state = { completed: false };

	handleChange = e => {
		const { name, type, value } = e.target;
		this.setState({ [name]: value });
	};

	updateItem = async (e, updateItemMutation) => {
		e.preventDefault();
		const res = await updateItemMutation({
			variables: {
				id: this.props.id,
				...this.state
			}
		});
		this.setState({ completed: true });
	};

	handleDismiss = () => {
		this.setState({ completed: false });
	};

	render() {
		return (
			<Query query={ITEM_QUERY} variables={{ id: this.props.id }}>
				{({ data, loading, error }) => {
					if (error) return <p>Error...</p>;
					return (
						<AddItemStyles>
							<Head>
								<title>Grouper | Edit {data.listItem.name}</title>
							</Head>
							<Grid container textAlign="center">
								<Grid.Column>
									<h1>Edit {data.listItem.name}</h1>
									<Link href={{ pathname: 'list', query: { id: data.listItem.list } }}>
										<Button
											inverted
											icon="left chevron"
											labelPosition="left"
											content="Return to List"
										/>
									</Link>
									<Mutation mutation={EDIT_ITEM_MUTATION} variables={this.state}>
										{(editListItem, { loading, error }) => {
											if (error) return <p>Error...</p>;
											return (
												<Form
													error={error}
													success={this.state.completed}
													inverted
													loading={loading}
													method="post"
													onSubmit={async e => {
														await this.updateItem(e, editListItem);
													}}>
													>
													<Message
														onDismiss={this.handleDismiss}
														success
														header={`${data.listItem.name} Updated`}
														content="Make further edits or go back to your list."
													/>
													<Message
														error
														header="Oops!"
														content={error && error.message.replace('GraphQL error: ', '')}
													/>
													<Form.Group>
														<Form.Input
															required
															width={13}
															type="text"
															name="name"
															id="name"
															label="Item"
															defaultValue={data.listItem.name || ''}
															minLength={5}
															maxLength={50}
															onChange={this.handleChange}
															placeholder="Enter Item Name - 50 Character Max"
														/>
														<Form.Input
															width={3}
															type="number"
															name="price"
															id="price"
															label="Max Price: 0 = No Max."
															defaultValue={data.listItem.price}
															onChange={this.handleChange}
															placeholder="Optional - Enter Whole Dollar Amounts"
															icon="dollar"
															iconPosition="left"
														/>
														<Button inverted icon="edit" className="add-item-button" />
													</Form.Group>
												</Form>
											);
										}}
									</Mutation>
								</Grid.Column>
							</Grid>
						</AddItemStyles>
					);
				}}
			</Query>
		);
	}
}

export default EditItem;
