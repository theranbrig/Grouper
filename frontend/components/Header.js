import Router from 'next/router';
import NProgress from 'nprogress';
import styled from 'styled-components';
import Nav from './Nav';
import Logo from './styles/Logo';
import StyledHeader from './styles/HeaderStyles';

Router.onRouteChangeStart = () => {
  NProgress.start();
};

Router.onRouteChangeComplete = () => {
  NProgress.done();
};

Router.onRouteChangeError = () => {
  NProgress.done();
};

const Header = () => (
  <StyledHeader>
    <Logo className="bar">
      <div>
        <img src="../static/logo.png" alt="Grouper Logo" />
      </div>
      <Nav />
    </Logo>
  </StyledHeader>
);

export default Header;
