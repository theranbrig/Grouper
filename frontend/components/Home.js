import Link from 'next/link';
import React, { Component } from 'react';
import { Button, Grid, Icon, Image } from 'semantic-ui-react';
import HomeStyles from './styles/HomeStyles';
import User from './User';

class Home extends Component {
	render() {
		return (
			<User>
				{({ data: { me } }) => {
					return (
						<HomeStyles>
							<Grid centered container>
								<Grid.Column mobile={16} tablet={12} computer={8} textAlign="center">
									<h1>Grouper</h1>
									<Image src="../static/whitefish.png" alt="Fish Logo" centered />
									<h2>Simple Group Shopping</h2>
									{me ? (
										<div className="button-area">
											<Button inverted size="big">
												<Link href="/lists">
													<a>Go to Lists</a>
												</Link>
											</Button>
										</div>
									) : (
										<div className="button-area">
											<Button inverted size="huge">
												<Link href="/login">
													<a>Login</a>
												</Link>
											</Button>
											<Button inverted size="big">
												<Link href="/join">
													<a>Sign Up</a>
												</Link>
											</Button>
										</div>
									)}
									<ul>
										<li>
                      <Icon name="group"/>
											<p>Divide and conquer while you shop. Split up and find what you need around the
											store.</p>
										</li>
										<li>
                      <Icon name="shopping cart"/>
											<p>Check off items as you pick them up. Let others know you found what you were
											looking for.</p>
										</li>
										<li>
                      <Icon name="dollar"/>
                      <p>Set price limits. Avoid overspending and stay within your budget.</p>
                    </li>
									</ul>
								</Grid.Column>
							</Grid>
						</HomeStyles>
					);
				}}
			</User>
		);
	}
}

export default Home;
