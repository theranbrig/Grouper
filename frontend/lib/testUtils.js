import casual from 'casual';

casual.seed(777);

const fakeUser = () => ({
  __typename: 'User',
  id: '4234',
  username: casual.name,
  email: casual.email,
  password: 'password',
  permissions: ['USER'],
});

const fakeItem = () => ({
  __typename: 'ListItem',
  id: '3534',
  name: 'Habaneros',
  price: 12,
  list: '1234567890',
  user: fakeUser(),
  inCart: false,
});

const fakeList = () => ({
  __typename: 'List',
  id: '01234567890',
  users: [fakeUser()],
  name: 'Grocery List',
  type: 'Groceries',
  items: [fakeItem()],
});

export { fakeList, fakeUser, fakeItem };
