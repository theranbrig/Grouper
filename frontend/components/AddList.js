import gql from 'graphql-tag';
import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import { Button, Form, Select } from 'semantic-ui-react';
import { typeOptions } from '../lib/formData';
import Error from './ErrorMessage';
import { LISTS_QUERY } from './Lists';
import FormStyles from './styles/FormStyles';

const CREATE_LIST_MUTATION = gql`
	mutation CREATE_LIST_MUTATION($name: String!, $type: String!) {
		createList(name: $name, type: $type) {
			id
			name
			type
		}
	}
`;

class AddList extends Component {
	state = {
		name: '',
		type: 'All Purpose'
	};

	saveToState = e => {
		this.setState({ [e.target.name]: e.target.value });
	};

	selectInput = (e, data) => {
		this.setState({ type: data.value });
	};

	render() {
		return (
			<div>
				<h2>Add New List</h2>
				<Mutation
					mutation={CREATE_LIST_MUTATION}
					refetchQueries={[{ query: LISTS_QUERY }]}
					variables={this.state}>
					{(createList, { error, loading }) => {
						if (error) return <Error error={error} />;
						return (
							<FormStyles>
								<Form
									inverted
									method="post"
									loading={loading}
									onSubmit={async e => {
										e.preventDefault();
										const res = await createList();
										this.setState({
											name: '',
											type: ''
										});
									}}>
									<Form.Group>
										<Form.Input
											width={12}
											type="text"
											name="name"
											label="List Name"
											value={this.state.name}
											onChange={this.saveToState}
											maxLength="40"
											placeholder="Enter List Name - 40 Char Max"
										/>
										<Form.Field
											width={4}
											control={Select}
											options={typeOptions}
											value={this.state.type}
											onChange={this.selectInput}
											label={{ children: 'Types', htmlFor: 'type' }}
											search
											fluid
											searchInput={{ id: 'type' }}
										/>
									</Form.Group>
									<Button inverted type="submit" size="medium" className="add-item-button">
										Add{loading ? 'ing' : ''} List
									</Button>
								</Form>
							</FormStyles>
						);
					}}
				</Mutation>
			</div>
		);
	}
}

export default AddList;
