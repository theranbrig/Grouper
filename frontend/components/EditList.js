import gql from 'graphql-tag';
import Link from 'next/link';
import React, { Component } from 'react';
import { Mutation, Query } from 'react-apollo';
import { Button, Form, Grid, Select, Message, Icon } from 'semantic-ui-react';
import { typeOptions } from '../lib/formData';
import Error from './ErrorMessage';
import { INDIVIDUAL_LIST_QUERY } from './List';
import FormStyles from './styles/FormStyles';

const EDIT_LIST_MUTATION = gql`
	mutation EDIT_LIST_MUTATION($id: ID!, $name: String, $type: String) {
		editList(id: $id, name: $name, type: $type) {
			id
			name
			type
		}
	}
`;

class EditList extends Component {
	state = { completed: false };

	handleChange = e => {
		const { name, type, value } = e.target;
		this.setState({ [name]: value });
	};

	selectInput = (e, data) => {
		this.setState({ type: data.value });
	};

	updateList = async (e, updateListMutation) => {
		e.preventDefault();
		console.log(this.state);
		const res = await updateListMutation({
			variables: {
				id: this.props.id,
				...this.state
			}
		});
		this.setState({ completed: true });
	};

	render() {
		return (
			<Query query={INDIVIDUAL_LIST_QUERY} variables={{ id: this.props.id }}>
				{({ data, loading, error }) => {
					console.log(data);
					if (error) return <p>Error...</p>;
					return (
						<FormStyles>
							<Grid container textAlign="center">
								<Grid.Column>
									<h1>
										<Icon name="edit" />
										Edit {data.list.name}
									</h1>
									<Link href={{ pathname: 'lists' }}>
										<Button
											inverted
											icon="left chevron"
											labelPosition="left"
											content="Back to Lists"
											className="back-link"
										/>
									</Link>
									<Mutation mutation={EDIT_LIST_MUTATION} variables={this.state}>
										{(editList, { error, loading }) => {
											if (error) return <Error error={error} />;
											return (
												<Form
													success={this.state.completed}
													inverted
													method="post"
													loading={loading}
													onSubmit={async e => {
														await this.updateList(e, editList);
													}}>
													<Message
														success
														header={`${data.list.name} Updated`}
														content="Make further edits or go back to your list."
													/>
													<Form.Group>
														<Form.Input
															width={12}
															type="text"
															name="name"
															label="List Name"
															defaultValue={data.list.name}
															onChange={this.handleChange}
															maxLength="40"
															placeholder="Enter List Name - 40 Char Max"
														/>
														<Form.Field
															width={4}
															control={Select}
															options={typeOptions}
															defaultValue={data.list.type}
															onChange={this.selectInput}
															label={{ children: 'Types', htmlFor: 'type' }}
															search
															fluid
															searchInput={{ id: 'type' }}
														/>
													</Form.Group>
													<Button inverted type="submit" size="medium" className="edit-list-button">
														Edit{loading ? 'ing' : ''} List
													</Button>
												</Form>
											);
										}}
									</Mutation>
								</Grid.Column>
							</Grid>
						</FormStyles>
					);
				}}
			</Query>
		);
	}
}

export default EditList;
