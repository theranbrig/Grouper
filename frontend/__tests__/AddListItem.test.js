import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import AddListItem from '../components/AddListItem';
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser } from '../lib/testUtils';

const mocks = [
	{
		request: { query: CURRENT_USER_QUERY },
		result: { data: { me: fakeUser() } }
	}
];

describe('<AddList/>', () => {
	it('renders and matches the snap shot', async () => {
		let apolloClient;
		const wrapper = mount(
			<MockedProvider mocks={mocks}>
				apolloClient = client;
				<AddListItem id="abc123" />
			</MockedProvider>
		);
		await wait();
		wrapper.update();
		expect(toJSON(wrapper.find('h2'))).toMatchSnapshot();
		expect(toJSON(wrapper.find('Button'))).toMatchSnapshot();
	});
});
