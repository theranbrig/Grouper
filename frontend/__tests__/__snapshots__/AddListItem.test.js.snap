// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddList/> renders and matches the snap shot 1`] = `null`;

exports[`<AddList/> renders and matches the snap shot 2`] = `
<Button
  as="button"
  className="add-item-button"
  icon="add"
  inverted={true}
  role="button"
>
  <button
    className="ui icon inverted button add-item-button"
    onClick={[Function]}
    role="button"
  >
    <Icon
      as="i"
      name="add"
    >
      <i
        aria-hidden="true"
        className="add icon"
      />
    </Icon>
  </button>
</Button>
`;
